firmware-src/sources/webserver.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * webserver.c
+ *
+ * Copyright 2016 DeviceHive
+ *
+ * Author: Nikolay Khabarov
+ *
+ */
+
+#include <c_types.h>
+#include <osapi.h>
+#include "webserver.h"
+#include "httpd.h"
+#include "rest.h"
+
+LOCAL int ICACHE_FLASH_ATTR check_rest(HTTP_RESPONSE_STATUS *res, const char *path,
+		const char *key, HTTP_CONTENT *content_in, HTTP_ANSWER *answer) {
+	static const char api[] = "/api";
+	if(os_strncmp(path, api, sizeof(api) - 1) == 0) {
+		const char *p = &path[sizeof(api) - 1];
+		if(p[0] == 0) {
+			*res = rest_handle(p, key, content_in, answer);
+			return 1;
+		}else if(p[0] == '/') {
+			*res = rest_handle(&p[1], key, content_in, answer);
+			return 1;
+		}
+	}
```

Comment:
No, we should. This method just checks if we handle RESTful request. Something like /apiABC in this case will be passed follow to the local static web page chooser which would respond with corresponding page or 404.